### ISSUE TITLE
New initiatives opened from gForm should create a PR with the details, as if it was opened from the new_initiative issue template

### ISSUE THREAD

**Issue by DanielShterenberg** [4tals/LinksForIsrael#484]
This is closing the loop for the new initiatives. 
Depends on #393

**Comment by ohadschn** [comment]
@DanielShterenberg I think this issue is redundant.
The DOD of https://github.com/4tals/LinksForIsrael/issues/393 should be opened issues that are indistinguishable from user-opened new-initiative issues (same title prefix, same labels, same markdown format). Once that is achieved, the problem is already solved.

**Comment by DanielShterenberg** [comment]
#393 is just the connection between the sheets and an issue. 
There is a minor minor difference between the two: I need to apply the labels as well. I already fixed it, but needed a 2nd iteration. 
I wrote this issue on a paper so decided to move it here is well.

Anyway, its going to be closed soon.

**Comment by ohadschn** [comment]
@DanielShterenberg so again, if

> There is a minor minor difference between the two: I need to apply the labels as well. 

The way I see it, this means the other issue was not completed, I don't think a new issue is required, we have too many of those (due to some guy who keeps opening them)

**Comment by DanielShterenberg** [comment]
Opening an issue is not opening a PR with the correct code ;)

Anyway, I opened it for myself and assigned myself on this so I could throw away my paper. All good.

I try to open as few tickets as I can, believe me.

**Comment by ohadschn** [comment]
OK, now that the other one is closed we can repoen this one 😏

IMO the opened issue should be indistinguishable from a user-opened issue (via the yaml form).
Here are the differences I see currently:

1. The titles don't match (e.g. `WhatsApp Link` vs `Initiative WhatsApp`)
2. `Category` appears twice 
3. The category is not one of the values we defined, per: https://github.com/4tals/LinksForIsrael/blob/e2be880d601ce66b9d3049ac17113448f8ef7867/.github/ISSUE_TEMPLATE/new_initiative.yml?plain=1#L50
4. Missing `[NEW-INITIATIVE]:` prefix
5. The issue's title should ideally be in Hebrew (and the Initiative Name in English)
6. Missing some fields - most importantly long description (AKA `Initiative Details`), but also LinkedIn and others
7. Empty response value should be `No response` (not `N/A`)
8. Twitter / X link should be a full clickable link (not just handle like `@4israelproject`)

**Comment by DanielShterenberg** [comment]
I updated the title of the issue so adjust it to your requests.
As discussed in the group-chat yesterday, we will need to adjust all the names (gForm, gSheet and templates).

**Comment by DanielShterenberg** [comment]
@4tal do we ha

**Comment by DanielShterenberg** [comment]
@ohadschn thanks for this comment. There are things that are easy to fix, such as:
> 1. The titles don't match in name (e.g. `WhatsApp Link` vs `Initiative WhatsApp`)

Done. Also renamed other titles.

> 2. The titles don't match in order (e.g. `Main Link` before `Description`)

Done. 

> 3. `Category` appears twice

Done.

> 5. Missing `[NEW-INITIATIVE]:` prefix

Done.

> 8. Empty response value should be `No response` (not `N/A`)

Done.

----
On the other hand, there are things that should be aligned between the form, yml and script:
> 4. The category is not one of the values we defined, per: https://github.com/4tals/LinksForIsrael/blob/e2be880d601ce66b9d3049ac17113448f8ef7867/.github/ISSUE_TEMPLATE/new_initiative.yml?plain=1#L50
> 7. Missing some fields - most importantly long description (AKA `Initiative Details`), but also LinkedIn and others (+also what @4tal mentioned).

----
And finally we have points that we should decide how do we want to handle them such as:
> 6. The issue's title should ideally be in Hebrew (and the Initiative Name in English)
> 9. Twitter / X link should be a full clickable link (not just handle like `@4israelproject`)

For these two, what should we do? Its a user input; Should we add instructions to the form or should we add code that handles that (over eng?)

**Comment by ohadschn** [comment]
> @4tal we dont have initiativeImage, initiativeValidationDetails in the form. How should we add it? Until its not there, we can't use it in the sheet and therefor we can't add it in the script

Just add it to the Gform and use it in the issue creation script only if it's there (basically treat it as an optional value).

> should be aligned between the form, yml and script

Same thing - update the form and script. For old initiatives that were already inserted with bad categories, leave as is and hope GPT does the right thing - if not, manually correct in the generated PR.

> Should we add instructions to the form or should we add code that handles that (over eng?)

For sure add instructions to the form, and when the instcution wasn't followed, hope GPT does the right thing (additional comments could be provided to help it along) - worst case, fix manually. We could also check simple cases only (e.g. for X/Twitter pattern match for `@x`), but it's not critical.

**Comment by DanielShterenberg** [comment]
> BTW, how come all recent initiative PRs have "suggested by daniel?
Can you pull the gmail alias of the person who actually made the suggestion?
Or if not, at least change it to something like "google form user"?

Its because the script uses my Github token.

I mentioned it to @4tal that we need to open an "Admin/CI" account for all general initiatives and use its credentials.

**Comment by DanielShterenberg** [comment]
@ohadschn, @4tal 

- What values are we expected to have for the `InitiativeImage`? A URL? A multipart-upload of an image? 
Also, I couldn't find an issue with such image, can you please provide an example?

- `initiativeValidationDetails` - What should be the title in the form? What is it for?

**Comment by ohadschn** [comment]
> Its because the script uses my Github token.
> 
> I mentioned it to @4tal that we need to open an "Admin/CI" account for all general initiatives and use its credentials.

And then what? The PR would be suggested by "Admin"? 
Still not great, I think we should:
1. Add optional submitter name field to the form and use that (the docs suggest there's no user data in this callback to take automatically: https://developers.google.com/apps-script/guides/triggers/events#form-submit)
2. If that field isn't filled, fall back to some value like "Google Forms User"

In other words, the GitHub token should have nothing to do with this.

> What values are we expected to have for the `InitiativeImage`? A URL? A multipart-upload of an image?
Also, I couldn't find an issue with such image, can you please provide an example?

Yes, URL. In the GitHub issue we call it `logo`, example: https://github.com/4tals/LinksForIsrael/issues/505

> `initiativeValidationDetails` - What should be the title in the form? What is it for?
Same as the issue form yaml: `Initiative Validation Details` (you can look at the sample issue provided above, it has it as well)
